proxy/client.go

```diff
@@ -0,0 +1,226 @@
+package proxy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/rollkit/go-execution"
+	proxy_json_rpc "github.com/rollkit/go-execution/proxy/jsonrpc"
+	rollkitTypes "github.com/rollkit/rollkit/types"
+)
+
+type PayloadStatus string
+
+const (
+	PayloadStatusValid   PayloadStatus = "VALID"
+	PayloadStatusInvalid PayloadStatus = "INVALID"
+	PayloadStatusSyncing PayloadStatus = "SYNCING"
+)
+
+var (
+	ErrNilPayloadStatus     = errors.New("nil payload status")
+	ErrInvalidPayloadStatus = errors.New("invalid payload status")
+)
+
+// Ensure ProxyClient implements Execute interface
+var _ execution.Execute = (*ProxyClient)(nil)
+
+// ProxyClient implements the Execute interface in go-execution
+type ProxyClient struct {
```

Comment:
IMHO this should be "top level" implementation; logic should be implemented directly in `EngineAPIExecutionClient`.